src/main/java/org/apache/cassandra/sidecar/utils/SSTableImporter.java

```diff
@@ -159,16 +159,45 @@ protected ImportQueue initializeQueue(String key)
      */
     private void processPendingImports(Long timerId)
     {
-        for (ImportQueue queue : importQueuePerHost.values())
+        for (Map.Entry<String, ImportQueue> entry : importQueuePerHost.entrySet())
         {
+            ImportQueue queue = entry.getValue();
             if (!queue.isEmpty())
             {
+                recordPendingImports(hostFromKey(entry.getKey()), entry.getValue().size());
                 executorPools.internal()
                              .executeBlocking(p -> maybeDrainImportQueue(queue));
             }
         }
     }
 
+    private void recordPendingImports(String host, int pendingImports)
+    {
+        if (host != null)
+        {
+            metadataFetcher.instance(host)
+                           .metrics()
+                           .sstableImport().pendingImports.metric.setValue(pendingImports);
+        }
+    }
+
+    /**
+     * Host is extracted from key for publishing Cassandra instance specific metrics related to {@link ImportQueue}
+     *
+     * @param key a key for one of the import queues
+     * @return host for which the import queue is maintained
+     */
+    protected String hostFromKey(String key)
+    {
+        String[] keyParts = key.split("\\$");
```

Comment:
do we really want to parse the key here? is there a better way to do this?

Reply:
As discussed will capture the metric while draining the queue